### DISCUSSION TITLE
Make some functions work the same way as on KolourPaint

### DISCUSSION THREAD

**Discussion by rado84-github** [PintaProject/Pinta#743] [Ideas]
1. On KolourPaint when you paste an image from file, the are automatically expands and becomes the size of the image. For instance, if you open KP and the initial field size has been 800x600 but you paste an image which is 4096x4096, the image field automatically expands to that size.
On Pinta the image field is 800x600 by default and DOESN'T expand when you import a larger image. So make the field to automatically expand when you import a larger image.
2. On KP when you click on the text button, you can not only resize the text field as much as you want in any direction but you can also grab the border of the text field and drag & drop that text wherever you want. Pinta has 2 movement tools but they move other things, not the text you've entered or its borders and they certainly DON'T move the text while you're entering it without affecting the background, so it would be great if Pinta had that function!
3. On KP you have these two buttons where you can choose between "transparency" and "opacity" and they apply not only for the text field (whether or not the text field to have a background) but also for imported images. That's a very cool function Pinta could use as well!
4. "Paste from file" - a function on KP which simply imports an image WITHOUT making it a layer. Pinta makes it a layer which isn't always convenient. It's good to have a layer function as well but 
5. And finally, it would be great if we could change the icon and the name of every single tool on Pinta as well as whether or not the tools to have an icon only, text only, or both, or text under/next to the icon. On KP if you right click on any button, there's this option "Tools settings" where you can choose an icon of your own for each tool.

You probably wonder why I'm suggesting these things. I've been using KP & Pinta in the past 8 or 9 years - Pinta for some things, KolourPaint for others. But QT keep breaking stuff and functions and frankly I'm tired of constantly trying to find fixes for their incompetency. If Pinta had these functions, KolourPaint would become truly obsolete because Pinta has functions I've been asking KDE to add to KP for years and guess whether they did it...

I could make a video about each suggested function, if there's anything unclear in the things I wrote.

**Comment by cameronwhite** [comment]
For the first suggestion, which method are you using to paste the image? When using Edit -> Paste or Ctrl+V to paste a larger image, this should ask you whether you want to expand the canvas
